efi/secureboot.go

```diff
@@ -49,6 +49,38 @@ type SignatureDBUpdate struct {
 	Data []byte                 // The update payload
 }
 
+type signatureDBUpdatesOption []*SignatureDBUpdate
+
+func (u signatureDBUpdatesOption) applyOptionTo(gen *pcrProfileGenerator) {
+	gen.varModifiers = append(gen.varModifiers, func(rootVars *rootVarsCollector) error {
+		for _, root := range rootVars.PeekAll() {
+			for _, quirk := range []signatureDBUpdateFirmwareQuirk{
+				signatureDBUpdateNoFirmwareQuirk,
+				signatureDBUpdateFirmwareDedupIgnoresOwner} {
+				// create branch by copying the root varBranch
+				branch := *root
```

Comment:
I'm probably missing something, how does the change propagate out of this function if below we operate on a copy?

Reply:
The copy maintains a copy of updates, and then each call to `WriteVar` propagates those updates to `rootVarsCollector` via its `registerUpdates` callback.

Reply:
I see maybe the comment should be:

// create a branch per quirk by copying the root varBranch

Reply:
for a moment I was confused and thought that root.updates was a slice, that is not case, otherwise the shallow copy would not be safe, as append to the same slice again can produce confusing results depending on the allocation history of the slice. I don't think we have code here that shallow copy structs with slices in them?